include/ccmath/math/nearest/ceil.hpp

```diff
@@ -8,7 +8,75 @@
 
 #pragma once
 
+#include "ccmath/math/compare/isinf.hpp"
+#include "ccmath/math/compare/isnan.hpp"
+
+#include <limits>
+#include <type_traits>
+
 namespace ccm
 {
+    /**
+     * @brief Computes the smallest integer value no lesser than num.
+     * @tparam T The type of the number.
+     * @param num A floating-point or integer value.
+     * @return If no errors occur, the smallest integer value no lesser than num, that is ⌈num⌉, is returned.
+     */
+    template <typename T>
+    constexpr T ceil(T num) noexcept 
+    {
+        if constexpr (std::is_floating_point_v<T>)
```

Comment:
If we apply the approach suggested with enabling this function if we are handling a float then this if constexpr is not necessary.